inc/classes/busting/trait-file-busting.php

```diff
@@ -130,35 +129,19 @@ private function get_all_files() {
 			return false;
 		}
 
-		$dir = $this->filesystem->dirlist( $dir_path );
-
-		if ( false === $dir ) {
-			Logger::error(
-				'Could not get the directory contents.',
-				[
-					self::LOGGER_CONTEXT,
-					'path' => $dir_path,
-				]
-			);
-			return false;
-		}
-
-		if ( ! $dir ) {
-			return [];
-		}
+		$pattern = '/' . sprintf(
+			$this->escape_file_name( $this->file_name_pattern ),
+			'([a-f0-9]{32}|local)'
+		) . '/';
 
-		$list    = [];
-		$pattern = $this->escape_file_name( $this->filename );
-		$pattern = \sprintf( $pattern, '(?<version>(?:[a-f0-9]{32}|local))' );
+		$entries = rocket_get_dir_files_by_regex( $dir_path, $pattern );
 
-		foreach ( $dir as $entry ) {
-			if ( 'f' !== $entry['type'] ) {
-				continue;
-			}
+		$list = [];
+		foreach ( $entries as $entry ) {
+			$file_name = $entry->getFilename();
 
-			if ( \preg_match( '/^' . $pattern . '$/', $entry['name'], $matches ) ) {
-				$list[ $entry['name'] ] = $matches['version'];
-			}
+			preg_match( $pattern, $file_name, $file_details_match );
+			$list[ $file_name ] = $file_details_match[1];
```

Comment:
What if there are no matches?

Reply:
it isn't the case here, at the following line
https://github.com/wp-media/wp-rocket/blob/0833d81e74efa7ea30a255f5b4adb878221779b0/inc/classes/busting/trait-file-busting.php#L137
we are searching for files with this pattern and then we are using preg_match with the same pattern to get the file version from the file name
For example:-
if we have a file named: `gtm-123456789abc.js` so the version is `123456789abc` and we got that from applying the pattern `gtm-([a-f0-9]{32}|local).js` on the filename.
So when reaching this line we must have one match.

Reply:
What you're saying then is the iterator will only return filesystem entries that have the file version.

Can we ever imagine a scenario where a match can't be made, where it fails for some reason?

I'm asking this because if that scenario, even a remote weird edge case, would cause a fatal error. If we could imagine an edge case, then we should guard against it to prevent the failure.

Reply:
Ok, I did the check now.